content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,273 @@
+- Data
+- Python environment with all required libraries.
+
+In addition to that you will need an account with your cloud provider of choice.
+In this tutorial I'll be provisioning infrastructure on Amazon Web Services
+(AWS). You can create an AWS account yourself, or ask your DevOps team to
+provide you with one. Make sure to
+[store your authentication credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html)
+(`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`) in your system's environment
+variables. Once you do, we can start the move with the help of Terraform and the
+Iterative Provider.
+
+## What is Terraform?
+
+[Terraform](https://www.terraform.io) is an open-source infrastructure-as-code
+tool that you'll need to
+[download and install](https://www.terraform.io/downloads.html) for this
+tutorial. With Terraform, you can create a configuration file in which you
+declaratively describe what infrastructure you'd like to have. This means that
+you do not need to write the instructions on what exact steps need to be taken.
+Instead, you describe what your infrastructure should ultimately look like.
+Behind the scenes, Terraform will figure out what needs to be done. If you've
+cloned the repo, you'll find the `main.tf`file in the root of the project --
+that's where we will be configuring Terraform.
+
+## Terraform Provider Iterative
+
+Terraform can orchestrate a plethora of various resources for you, but for the
+majority of projects you only need a few. That's why instead of shipping all the
+integrations in one bundle, you install a barebones Terraform distribution and
+then plug in support for whatever resources you need with so-called
+[_providers_](https://www.terraform.io/docs/extend/how-terraform-works.html).
+For this tutorial we will only need
+[Iterative Provider](https://registry.terraform.io/providers/iterative/iterative/latest).
+It enables full lifecycle management of computing resources for machine learning
+pipelines from AWS, Microsoft Azure, Google Cloud Platform, and more. The
+Iterative Provider has a couple of advantages for machine learning pipelines.
+Namely,
+
+- The configuration for various cloud compute providers with the Iterative
+  Provider will be nearly identical, so you can easily migrate from one cloud
+  provider to another, if you want to.
+- It is designed not only to provision infrastructure but to execute your
+  scripts on it too, all via a single configuration.
+- It helps to sync data between your local machine and a remote one.
+- Once your training is complete, the remote resources will be terminated,
+  avoiding unused machines quietly ramping up costs.
+
+To start using the Iterative Provider we need to let Terraform know about it by
+adding the following in our `main.tf`:
+
+```hcl
+terraform {
+    required_providers { iterative = { source = "iterative/iterative", version = ">= 0.9.9" } }
+}
+provider "iterative" {}
+```
```

Comment:
Minor: Is it possible to break into more lines to avoid horizontal scrolling?

![image](https://user-images.githubusercontent.com/1477535/152391835-721002b5-3784-4913-afce-e62f0748dead.png)